src/CommunityToolkit.Maui.Core/CommunityToolkit.Maui.Core.csproj

```diff
@@ -6,12 +6,6 @@
     <UseMaui>true</UseMaui>
     <SingleProject>true</SingleProject>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
-
-    <SupportedOSPlatformVersion Condition="'$(TargetFramework)' == 'net6.0-ios'">14.2</SupportedOSPlatformVersion>
```

Comment:
I think we need to keep this code
1. it is included in MAUI templates
2. It validates that we do not use unsupported APIs (e.g. Android 19)
3. By removing this you remove symptoms but not the root cause of the issue. I don't know exactly, but let's say Visual Studio/MSBuild incorrectly resolves the target framework

Reply:
I believe this is only meant for Apps, not Libraries. 

If I understand it correctly, it's the same thing as setting the target framework in the AndroidManifest.xml or the Info.plist.

Neither the [.NET MAUI Essentials CSPROJ](https://github.com/dotnet/maui/blob/main/src/Essentials/src/Essentials.csproj) nor the [.NET MAUI CSPROJ](https://github.com/dotnet/maui/blob/main/src/Core/src/Core.csproj) include `SupportedOSPlatformVersion` and `TargetPlatformMinVersion`.

Reply:
But MAUI lib includes it: https://github.com/dotnet/maui/blob/main/src/Templates/src/templates/maui-lib/MauiLib1.csproj

as we can see it is not only for AndroidManifest.xml or Info.plist, it also validates the compatibility code and the target device.

Reply:
Right - [MauiLib1.csproj](https://github.com/dotnet/maui/blob/main/src/Templates/src/templates/maui-lib/MauiLib1.csproj) is template is for Apps, not libraries.

We need to keep these values for our Sample App, `CommunityToolkit.Maui.Sample`, but will remove it from our libraries, `CommunityToolkit.Maui`, `CommunityToolit.Maui.Core` and `CommunityToolkit.Maui.Markup`.